jax/_src/random.py

```diff
@@ -572,7 +586,7 @@ def choice(key, a, shape=(), replace=True, p=None):
     a = int(a)
   else:
     a = _asarray(a)
-  n_inputs = a if np.ndim(a) == 0 else len(a)
+  n_inputs = int(a) if np.ndim(a) == 0 else len(a)  # type: ignore[arg-type]
```

Comment:
This cast ensures that from here on `n_inputs` is an `int` (not an `np.int32` or a `jnp.array` of shape `()`)